### PR TITLE
Allow user to clear memory of an agent

### PR DESCRIPTION
**Pull request by knu**
This addresses #802.

What do you guys think about this new CSS animated spinner, by the way?

### CODE REVIEW

**Comment by virtadpt** [comment]
Where is the animated spinner utilized?

File: spec/controllers/agents_controller_spec.rb [hunk 1]
```diff
@@ -374,4 +374,14 @@ def valid_attributes(options = {})
       }
     end
   end
+
+  describe "DELETE memory" do
```
Comment (cantino) [inline spec/controllers/agents_controller_spec.rb]: Might be good to add a defensive test showing that you cannot delete the memory of some other user's agent?

**Comment by cantino** [comment]
Nice @knu, I like it!

File: spec/controllers/agents_controller_spec.rb (hunk 1 above)
Comment (knu) [inline spec/controllers/agents_controller_spec.rb]: Sure, done!

**Comment by cantino** [comment]
Nice!

**Pull request merged**